src/lib/components/Treebank/Text/Text.js

```diff
@@ -7,9 +7,15 @@ import Text from '../../Text';
 const TextWithContext = () => (
   <SentenceContext.Consumer>
     {({
-      sentence, active, toggleActive, config,
+      sentence, active, toggleActive, highlighted, config,
```

Comment:
It's a little confusing that there's a `src/lib/components/Treebank/Text/Text.js` file and a `src/lib/components/Text/Text.js` file.

The reason for this goes back to React Contexts (mentioned above). `src/lib/components/Text/Text.js` is just a regular component, that does not use a Context. This makes it much easier to test because all of the values used within the component are passed in explicitly:

```jsx
<Text highlighted={new Set(['1', '2'])} />
```

---

The `src/lib/components/Treebank/Text/Text.js` imports the context, and the `src/lib/components/Text/Text.js` component and returns a component whose values are taken from the context:

```jsx
<Sentence highlighted={['1', '2']}>
  <Text />
</Sentence>
```

Reply:
If to be honest - it is really confusing to see this:

![image](https://user-images.githubusercontent.com/12377640/111617299-a8fda380-882e-11eb-816d-bb9c9bc46bef.png)

and the same inside Treebank
![image](https://user-images.githubusercontent.com/12377640/111617334-b450cf00-882e-11eb-9891-81059d03e286.png)

I don't know React well, but do you really need this duplicated structure?

Reply:
This structure looks confusing, but I think it's the best way to organize code using React Contexts. Consider two hypothetical components: one of them displays a list of highlighted words, the other one shows grammatical information related to the active word. We would use them in an app like this:

```jsx
<Sentence active="4" highlight={["1", "3", "10"]}>
  <HighlightedWords />
  <ActiveWordGrammar />
</Sentence>
```

The `<Sentence>` component passes `active` and `highlight` to the children. But now, let's say we wanted to test the `<HighlightedWords>` component. We would have to write the tests like this:

```jsx
<Sentence active="4" highlight={["1", "3", "10"]}>
  <HighlightedWords />
</Sentence>
```

In order to test `<HighlightedWords>`, we would have to create a whole `<Sentence>` component and everything it requires. But this is a problem. We're adding a bunch of things to our tests unrelated to the functionality that we actually want to test. When testing `<HighlightedWords>`, we shouldn't have to care about `<Sentence>`, and we definitely shouldn't have to care about `active`. 

To address this, we create a separate `<HighlightedWords>` component that does not rely on `<Sentence>`. We can now test it more easily:

```jsx
<HighlightedWords highlight={["1", "3", "10"]} />
```

Doing it this way makes testing easier, but more importantly, it makes the code easier to reason about (making things easier to test often has this side effect). Each component is much simpler and only uses the props given to it.

The only problem is that we have to now create a component that connects the `<Sentence>` and `<HighlightedWords>`. This makes the code look duplicated, but these components are mostly bookkeeping and don't add any functionality of their own.

Reply:
I am not well-familiar with React and couldn't suggest any other way to organize in such cases.
There are several approaches on this points - should we test the code that is currently used in the application or should we make some adaptations for testing. This is a question what is the priority - test-driven development or first is optimized development and then testing.
I like more independent design in the code and try to avoid global parameters. 
But I am not sure if it is the same situation here. And anyway your code is easy to read - so it is well-organized internally :)

May be it is worth to add such an explanation somewhere in readme for tests?